src/main/java/in/zapr/druid/druidry/query/scan/DruidScanQuery.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2017-present, Red Brick Lane Marketing Solutions Pvt. Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package in.zapr.druid.druidry.query.scan;
+
+import com.google.common.base.Preconditions;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.List;
+
+import in.zapr.druid.druidry.Context;
+import in.zapr.druid.druidry.Interval;
+import in.zapr.druid.druidry.dimension.DruidDimension;
+import in.zapr.druid.druidry.filter.DruidFilter;
+import in.zapr.druid.druidry.query.DruidQuery;
+import in.zapr.druid.druidry.query.QueryType;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+
+@Getter
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode(callSuper = true)
+public class DruidScanQuery extends DruidQuery {
+
+    private DruidFilter filter;
+    private Integer batchSize;
+    private List<Interval> intervals;
+    private List<DruidDimension> columns;
```

Comment:
1. Value can be dimensions and metrics. Using DruidDimension would seem restricting or if metric, needed to be added, put metrics in dimension object, which increases the hackiness of code and also, decrease readability.
2. It can be a string array. DruidDimension can be a complex object depending on sub-class being used.

I think it should be List<String>.
Ref: http://druid.io/docs/latest/querying/scan-query.html